archiver.go

```diff
@@ -8,6 +8,8 @@ import (
 	"path/filepath"
 	"runtime"
 	"strings"
+
+	"github.com/cyphar/filepath-securejoin"
```

Comment:
I'm not a huge fan of introducing a whole dependency just for one function... is the logic something we can extract from it pretty easily?

Reply:
shouldn't be too hard, I guess. It's 135 loc including comments and whitespaces - https://github.com/cyphar/filepath-securejoin/blob/master/join.go
I would, however, consider using the lib since it's not as simple a task as it seems. (it's the only thing it's doing, btw, so we get almost no unnecessary code).

Reply:
bump @mholt